lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -0,0 +1,500 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using Org.Apache.REEF.Common;
+using Org.Apache.REEF.Driver.Context;
+using Org.Apache.REEF.Driver.Evaluator;
+using Org.Apache.REEF.Driver.Task;
+using Org.Apache.REEF.IMRU.OnREEF.Driver.StateMachine;
+using Org.Apache.REEF.Network.Group.Driver;
+using Org.Apache.REEF.Tang.Implementations.Configuration;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Utilities.Attributes;
+using Org.Apache.REEF.Utilities.Diagnostics;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver
+{
+    /// <summary>
+    /// Manages Tasks, maintains task states and responsible for task submission
+    /// It covers the functionality in TaskStarter which will be not used in IMRU driver. 
+    /// </summary>
+    [NotThreadSafe]
+    internal sealed class TaskManager
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(TaskManager));
+
+        internal const string TaskAppError = "TaskAppError";
+        internal const string TaskSystemError = "TaskSystemError";
+        internal const string TaskGroupCommunicationError = "TaskGroupCommunicationError";
+        internal const string TaskEvaluatorError = "TaskEvaluatorError";
+        internal const string TaskKilledByDriver = "TaskKilledByDriver";
+        internal const string CloseTaskByDriver = "CloseTaskByDriver";
+
+        private readonly IDictionary<string, Tuple<TaskStateMachine, IConfiguration, IActiveContext>> _tasks
+            = new Dictionary<string, Tuple<TaskStateMachine, IConfiguration, IActiveContext>>();
+        private readonly IDictionary<string, IRunningTask> _runningTasks = new Dictionary<string, IRunningTask>();
```

Comment:
Yes. After tasks are running, we may need to close Running tasks when failure happens. In that case, we need the RunningTask references to call method on it.